### PR TITLE
Add chromatic plasma lens model

### PR DESCRIPTION
**Pull request by cemitch99**
Added a new element corresponding to an active (linear) plasma lens with chromatic focusing.  

This is equivalent to the quadrupole model in ChrQuad, except that the sign of the focusing in Y is reversed, so the model is focusing in both X and Y simultaneously.

- [x] add plasma lens element
- [x] add Python equivalents
- [x] add example from Lindstrom
- [x] update documentation
- [x] add updated test analysis script

### CODE REVIEW

File: examples/apochromatic/README.rst [hunk 1]
```diff
@@ -56,3 +56,63 @@ We run the following script to analyze correctness:
    .. literalinclude:: analysis_apochromatic.py
       :language: python3
       :caption: You can copy this file from ``examples/apochromatic/analysis_apochromatic.py``.
+
+
+.. _examples-apochromat_pl:
+
+Apochromatic Drift-Plasma Lens Beamline
+========================================
```
Comment (ax3l) [inline examples/apochromatic/README.rst]:
```suggestion
Apochromatic Drift-Plasma Lens Beamline
=======================================
```

File: examples/apochromatic/README.rst [hunk 2]
```diff
@@ -56,3 +56,63 @@ We run the following script to analyze correctness:
...
+* :math:`\alpha_\mathrm{x} = 0`
+* :math:`\beta_\mathrm{y} = 0.325` m
+* :math:`\alpha_\mathrm{y} = 0`
+
+We use a 100 GeV electron beam with an initially 6D Gaussian distribution of normalized rms emittance 1 micron and relative energy spread of 1%.
+
+The second moments of the particle distribution after the focusing beamline should coincide with the second moments of the particle distribution before the beamline, to within the l>
+The emittance growth due to chromatic effects remain below 1%.  In the absence of chromatic correction, the projected emittance growth is near 10%.
+
+In this test, the initial and final values of :math:`\sigma_x`, :math:`\sigma_y`, :math:`\sigma_t`, :math:`\epsilon_x`, :math:`\epsilon_y`, and :math:`\epsilon_t` must agree with no>
```
Comment (ax3l) [inline examples/apochromatic/README.rst]:
```suggestion
In this test, the initial and final values of :math:`\sigma_x`, :math:`\sigma_y`, :math:`\sigma_t`, :math:`\epsilon_x`, :math:`\epsilon_y`, and :math:`\epsilon_t` must agree with nominal values.
```

File: examples/apochromatic/README.rst [hunk 3]
```diff
@@ -56,3 +56,63 @@ We run the following script to analyze correctness:
...
+The matched Twiss parameters at entry are:
+
+* :math:`\beta_\mathrm{x} = 0.325` m
+* :math:`\alpha_\mathrm{x} = 0`
+* :math:`\beta_\mathrm{y} = 0.325` m
+* :math:`\alpha_\mathrm{y} = 0`
+
+We use a 100 GeV electron beam with an initially 6D Gaussian distribution of normalized rms emittance 1 micron and relative energy spread of 1%.
+
+The second moments of the particle distribution after the focusing beamline should coincide with the second moments of the particle distribution before the beamline, to within the l>
```
Comment (ax3l) [inline examples/apochromatic/README.rst]:
```suggestion
The second moments of the particle distribution after the focusing beamline should coincide with the second moments of the particle distribution before the beamline, to within the level expected due to noise due to statistical sampling.
```

File: examples/apochromatic/analysis_apochromatic_pl.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2023 ImpactX contributors
+# Authors: Axel Huebl, Chad Mitchell
+# License: BSD-3-Clause-LBNL
+#
+
```
Comment (ax3l) [inline examples/apochromatic/analysis_apochromatic_pl.py:7]:
```suggestion
```

File: examples/apochromatic/input_apochromatic_pl.in [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+beam.sigmaX = 1.288697604e-6
+beam.sigmaY = 1.288697604e-6
+beam.sigmaT = 1.0e-6
+beam.sigmaPx = 3.965223396e-6
+beam.sigmaPy = 3.965223396e-6
+beam.sigmaPt = 0.01  #1% energy spread
+beam.muxpx = 0.0
+beam.muypy = 0.0
+beam.mutpt = 0.0
+
```
Comment (ax3l) [inline examples/apochromatic/input_apochromatic_pl.in:19]:
```suggestion


```

File: examples/apochromatic/input_apochromatic_pl.in [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+q6.type = plasma_lens_chromatic
+q6.ds = 0.176038957633108457
+q6.g = 528.485181135649785
+#q6.k = 1.584350618697976110
+
+q7.type = plasma_lens_chromatic
+q7.ds = 0.331817852986604588
+q7.g = 996.147787384348956
+#q7.k = 2.98636067687944129
+
```
Comment (ax3l) [inline examples/apochromatic/input_apochromatic_pl.in:77]:
```suggestion


```

File: examples/apochromatic/run_apochromatic_pl.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2023 ImpactX contributors
+# Authors: Axel Huebl, Chad Mitchell
+# License: BSD-3-Clause-LBNL
+#
+# -*- coding: utf-8 -*-
+
+import amrex.space3d as amr
```
[the remark this answers is not in the document]
Comment (ax3l) [reply examples/apochromatic/run_apochromatic_pl.py]:
```suggestion
```

File: src/particles/elements/ChrPlasmaLens.H [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/alignment.H"
+#include "mixin/beamoptic.H"
+#include "mixin/thick.H"
+#include "mixin/nofinalize.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Print.H>      // for PrintToFile
```
Comment (ax3l) [inline src/particles/elements/ChrPlasmaLens.H]:
```suggestion
```

**Review by ax3l — APPROVED** [review]
Thank you, that looks excellent!

What is the difference with the `g` and `k` parameter left in the comments? Should we add this relation/conversion in the user-facing docs?

**Comment by cemitch99** [comment]
> Thank you, that looks excellent!
> 
> What is the difference with the `g` and `k` parameter left in the comments? Should we add this relation/conversion in the user-facing docs?

The `k` parameter in the comments is left over from copying the source from ChrQuad, and could be deleted.
However, after some thought maybe it is better to modify the input parameter to allow two sets of units, as we do for the quad, with the same symbol `k`.  Working on commits now so that ChrQuad and ChrPlasmaLens will use the same inputs.

File: docs/source/usage/parameters.rst [hunk 9]
```diff
@@ -367,7 +367,12 @@ Lattice Elements
            This requires these additional parameters:
 
             * ``<element_name>.ds`` (``float``, in meters) the segment length
-            * ``<element_name>.g`` (``float``, in T/m) the azimuthal magnetic field gradient in T/m
+            * ``<element_name>.k`` (``float``, in inverse meters squared OR in T/m) the plasma lens focusing strength
+                = (azimuthal magnetic field gradient in T/m) / (magnetic rigidity in T-m) - if units = 0
+
+             OR = azimuthal magnetic field gradient in T/m - if units = 1
+
+            * ``<element_name>.units`` (``integer``) specification of units (default: ``0``)
```
Comment (ax3l) [inline docs/source/usage/parameters.rst:375]: We can make input files more self-describing if we name the units, e.g.,
```
q3.units = "strength"
q4.units = "gradient"
```
or
```
q3.units = "1/m2"
q4.units = "T/m"
```
we can decide what would be a good naming and change this for more elements in a follow-up PR.

**Review by ax3l — APPROVED** [review]

**Review by n01r — COMMENTED** [review]
Awesome, thanks @cemitch99!

**Pull request merged**